GitHub V1 (#1)
stackhpc/stackhpc.kayobe_automation_workflows: roles/github/README.md

- `arguments`: list of arguments keys used by the automation task the contents will be acquired from `kayobe_arguments` or the defaults.

> - `use_bespoke`: Some workflows benefit from a dedicated workflow template as they drift away from the main template. Set to `true` if the workflow requires a *bespoke* template and ensure a template `workflow_name.yml.j2` is present.

Reviewer: Where does this template need to go?

> The following will override `workflows` to ensure only `Build Kayobe Image` and `Run Kolla Config Diff` is generated.

Reviewer: ```suggestion
The following will override `github_workflows` to ensure only `Build Kayobe Image` and `Run Kolla Config Diff` is generated.
```

```yaml
github_workflows:
  - "{{ build_kayobe_image }}"
  - "{{ run_kolla_config_diff }}"
```

Template Hooks
--------------

Workflows can be expanded with the use of hooks which are variables that if provided can be inserted into the appropriate location enabling the introduction of additional steps within the workflow job. This could include the use of HashiCorp Vault or installing and configuring a network proxy.

There are currently three hooks available

> - `github_checkout_hook`: a hook that occurs before the repository is cloned by the `checkout` action.

Reviewer: github_pre_checkout_hook?

Author: I have omitted the `pre` qualifier as I had removed the `post` hooks entirely which felt redundant.

The idea is each hook occurs before the named event. However, I can see the benefit in being clearer and can `pre` back.

> - `github_kayobe_hook`: a hook that occurs after the the repository has been cloned and before the kayobe automation task has started.

Reviewer: github_pre_kayobe_hook?

Author: See above regarding github_pre_checkout_hook?

- `github_final_hook`: a hook that occurs after the kayobe automation task has finished.

> A hook must be defined within the variables file for the playbook should be a scalar block string.

Reviewer: ```suggestion
A hook must be defined as a variable and should be a scalar block string.
```
